library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -1172,6 +1178,9 @@ private void clearRenderedFirstFrame() {
       MediaCodec codec = getCodec();
       // If codec is null then the listener will be instantiated in configureCodec.
       if (codec != null) {
+        if (tunnelingOnFrameRenderedListener != null) {
+          tunnelingOnFrameRenderedListener.destroyHandler(codec);
+        }
         tunnelingOnFrameRenderedListener = new OnFrameRenderedListenerV23(codec);
```

Comment:
If there is a listener installed, this leads to 2 calls to calls to `codec.setOnFrameRenderedListener`. One with null, the next one with the new listener.
This leads to disabling codec timestamp reporting then reenabling it immediately. Which is a costly operation as far as I can tell.
Thus I think that destoryHandler should not setOnFrameRenderedListener.
How about creating a function to install a new Listener:
```
MediaCodec.OnFrameRenderedListener oldListener = tunnelingOnFrameRenderedListener;
tunnelingOnFrameRenderedListener = new OnFrameRenderedListenerV23(codec);
if (oldListener != null) {
  oldListener.destroyHandler()
}
```
Maybe add a OnFrameRenderedListenerV23.disable method to codec.setOnFrameRenderedListener(null, null); in disable(). Although I'm not sure it is necessary.

Reply:
If you prefer, I can merge your PR as is and do a patch with this optimization over it.

Reply:
@stevemayhew, let me know if you would prefer do this change yourself or I can go along and do it on top of yours. We would like to merge your PR. Thanks

Reply:
Removing the old handler is just an optimization.  The check for stale events is sufficient.  I will add the function as you describe and the disable() function but not call it for now with a nite with your comments as to why.  Should have this today for you to review

Reply:
This commit 72f2e7d4fa6a29aa0cf6bc3a1bf2a8738b4f8ac0 adds the method and removes the double call to `codec.setOnFrameRenderedListener` if the codec is not being disabled.